templates/multistage-buildconfig.yaml

```diff
@@ -0,0 +1,99 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  annotations:
+    description: Template to produce an imagestream and buildconfig for a Jlinked image
+  name: jlinked-image-template
+objects:
+- apiVersion: image.openshift.io/v1
+  kind: ImageStream
+  metadata:
+    name: jlinked-image
+  spec:
+    lookupPolicy:
+      local: false
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: ubi9-jlinked-image-buildconfig
+  spec:
+    source:
+      dockerfile: |
+        #Stage-1: ubi9-jlinked-image is builder image + application + jlinked JVM
+        # install additional system dependencies (for ubi-micro) to /mnt/jrootfs
+        FROM -
+        USER 0
+        RUN mkdir -p /mnt/jrootfs
+        RUN microdnf install --installroot /mnt/jrootfs --releasever 9 --setopt install_weak_deps=0 --nodocs -y \
+            --config=/etc/dnf/dnf.conf \
+            --noplugins \
+            --setopt=cachedir=/var/cache \
+            --setopt=reposdir=/etc/yum.repos.d \
+            --setopt=varsdir=/etc/dnf/vars \
+            grep gawk
+        RUN rm -rf /mnt/jrootfs/var/cache/* /mnt/jrootfs/var/lib/rpm /mnt/jrootfs/var/lib/dnf
+      strategy:
+        dockerStrategy:
+          from:
+            kind: ImageStreamTag
+            name: quay.io/repository/jdowland/jlink:ubi9-jlinked-image
```

Comment:
This is a registry URI , but it needs to be an ImageStreamTag. The eventually-correct value for this will depend upon Josh's Phase 1 template. for now you could put `jlinked-image` which definitely exists (your template creates it) as a placeholder. I'd put a comment after `# placeholder value`